src/java/org/apache/cassandra/tools/nodetool/ClearSnapshot.java

```diff
@@ -43,34 +46,60 @@ public class ClearSnapshot extends NodeToolCmd
     @Option(title = "clear_all_snapshots", name = "--all", description = "Removes all snapshots")
     private boolean clearAllSnapshots = false;
 
+    @Option(title = "older_than", name = "--older-than", description = "Clear snapshots older than specified time period.")
+    private String olderThan;
+
+    @Option(title = "older_than_timestamp", name = "--older-than-timestamp", description = "Clear snapshots older than specified timestamp.")
+    private Long olderThanTimestamp;
```

Comment:
Why is this a long and not a isodate (`2022-09-18T12:09:54+0000`) ?

Reply:
No problem, I can change this to iso, the string provided would be parsed before calling jmx method so we know it is valid.

Reply:
Changed to String and parsing it to be sure it is valid Instant.